src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -154,6 +158,189 @@ def coe_add_monoid_hom : finite_measure α →+ measure α :=
+begin
+  apply is_finite_measure.lintegral_lt_top_of_bounded_to_ennreal,
+  use nndist f 0,
+  intros x,
+  have key := bounded_continuous_function.nnreal.upper_bound f x,
+  rw ennreal.coe_le_coe,
+  have eq : nndist f 0 = ⟨dist f 0, dist_nonneg⟩,
+  { ext,
+    simp only [real.coe_to_nnreal', max_eq_left_iff, subtype.coe_mk, coe_nndist], },
+  rwa eq at key,
+end
+
+@[simp] lemma test_against_nn_coe_eq {μ : finite_measure α} {f : α →ᵇ ℝ≥0} :
+  (μ.test_against_nn f : ℝ≥0∞) = ∫⁻ x, f x ∂(μ : measure α) :=
+ennreal.coe_to_nnreal (lintegral_lt_top_of_bounded_continuous_to_nnreal μ f).ne
+
+lemma test_against_nn_const (μ : finite_measure α) (c : ℝ≥0) :
+  μ.test_against_nn (bounded_continuous_function.const α c) = c * μ.mass :=
+by simp [← ennreal.coe_eq_coe]
+
+lemma test_against_nn_mono (μ : finite_measure α)
+  {f g : α →ᵇ ℝ≥0} (f_le_g : (f : α → ℝ≥0) ≤ g) :
+  μ.test_against_nn f ≤ μ.test_against_nn g :=
+begin
+  simp only [←ennreal.coe_le_coe, test_against_nn_coe_eq],
+  apply lintegral_mono,
+  exact λ x, ennreal.coe_mono (f_le_g x),
+end
+
+variables [opens_measurable_space α]
+
+lemma test_against_nn_add (μ : finite_measure α) (f₁ f₂ : α →ᵇ ℝ≥0) :
+  μ.test_against_nn (f₁ + f₂) = μ.test_against_nn f₁ + μ.test_against_nn f₂ :=
+begin
+  simp only [←ennreal.coe_eq_coe, bounded_continuous_function.coe_add, ennreal.coe_add,
+             pi.add_apply, test_against_nn_coe_eq],
+  apply lintegral_add,
+  repeat { exact (bounded_continuous_function.nnreal.to_ennreal_comp_measurable _), },
+end
+
+-- Hmmm, what is the right way to do this `nnreal_mul_ennreal_to_nnreal`?
+-- If this deserves to be added, then perhaps in `data.real.ennreal`?
+-- It seems like a typical coercion issue to me, although it is only used once here.
+lemma _root_.ennreal.nnreal_mul_ennreal_to_nnreal (c : ℝ≥0) (z : ℝ≥0∞) :
+  c * z.to_nnreal = ((c : ℝ≥0∞) * z).to_nnreal :=
+begin
+  by_cases z_infty : z = ∞,
+  { rw z_infty,
+    simp only [ennreal.top_to_nnreal, ennreal.to_nnreal_mul_top, mul_zero], },
+  { have z_lt_top : z < ∞ := ne.lt_top z_infty,
+    simp only [ennreal.to_nnreal_mul, ennreal.to_nnreal_coe], },
+end
+
+lemma test_against_nn_smul (μ : finite_measure α) (c : ℝ≥0) (f : α →ᵇ ℝ≥0) :
+  μ.test_against_nn (c • f) = c * μ.test_against_nn f :=
+begin
+  simp only [←ennreal.coe_eq_coe, algebra.id.smul_eq_mul, bounded_continuous_function.coe_smul,
+             test_against_nn_coe_eq, ennreal.coe_mul],
+  have key_smul := @lintegral_mul_const _ _ (μ : measure α) c _
+                   (bounded_continuous_function.nnreal.to_ennreal_comp_measurable f),
+  simp_rw mul_comm at key_smul,
+  exact key_smul,
+end
+
+/-- Integration against a finite_measure defines a linear map from nonnegative bounded continuous
+functions to nonnegative real numbers. -/
+def test_against_nn_linear_map (μ : finite_measure α) : (α →ᵇ ℝ≥0) →ₗ[ℝ≥0] ℝ≥0 :=
+{ to_fun := (λ f, μ.test_against_nn f),
+  map_add' := test_against_nn_add μ,
+  map_smul' := test_against_nn_smul μ, }
```

Comment:
Better to have fewer rather than more versions of a given object.  You can always use the coercion of `to_weak_dual_of_bounded_continuous_nnreal` in place of this, so I'd delete this.